[docs] Added .htaccess for static on Apache (#6625)
sveltejs/kit: packages/adapter-static/README.md

export const ssr = false;
```

> #### Apache Server
> 
> For a Single Page Application on an `Apache` Server, you need an `.htaccess` to route all of the requests into the entry file for the routing to work, otherwise it will throw a 404 error once you refresh or visit a page that isn't compiled.

Reviewer: ```suggestion
#### Apache

To run an SPA on [Apache](https://httpd.apache.org/), you need a `static/.htaccess` file to route requests to the fallback page:
```

```
<IfModule mod_rewrite.c>
  RewriteEngine On
  RewriteBase /

>   RewriteRule ^app\.html$ - [L]
>   RewriteCond %{REQUEST_FILENAME} !-f
>   RewriteCond %{REQUEST_FILENAME} !-d
>   RewriteRule . /app.html [L]

Reviewer: I assume `app.html` is just a fallback page. Given that the previous code snippet uses `200.html`, this one probably should too:

```suggestion
  RewriteRule ^200\.html$ - [L]
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule . /200.html [L]
```

Will this configuration allow Apache to serve pages that _were_ prerendered, if I have a mix of prerendered pages and ones that should be handled by the fallback?

Author: @Rich-Harris thank you for the suggestions!
Yes, I also created a sample about this build.
Here: http://svelte-on-apache.great-site.net

Reviewer: Thanks — it looks like while the prerendered index page is served correctly...

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/1162160/191758888-543a8d7d-5c7b-4e82-8719-b01b675319d2.png">

...the prerendered about page is blank:

<img width="812" alt="image" src="https://user-images.githubusercontent.com/1162160/191759056-0d54c144-5bd9-42ff-a6ba-1e08cdc806a7.png">

I have to visit `/about.html` to see the prerendered `/about` content (and ideally this URL would be rewritten to `/about`). Do you know how to edit the `.htaccess` to achieve that behaviour?

Author: Thank you! I updated the `.htaccess` to check if there's a html file.
Sample: http://svelte-on-apache.great-site.net/about

```
<IfModule mod_rewrite.c>
  RewriteEngine on
  RewriteBase /

  RewriteCond %{REQUEST_FILENAME}.html -f
  RewriteRule . %{REQUEST_FILENAME}.html [L]
	
  RewriteCond %{REQUEST_FILENAME} !-f
  RewriteCond %{REQUEST_FILENAME} !-d
  RewriteRule . 200.html [L]
</IfModule>
```

Ref: https://httpd.apache.org/docs/current/mod/mod_rewrite.html

Reviewer: Yeah, this is better. It's serving these URLs though, which is buggy:

* http://svelte-on-apache.great-site.net/about.html
* http://svelte-on-apache.great-site.net/about/

Both should be redirected to `/about` (unless `trailingSlash` was `'always'`, in which case SvelteKit would generate `about/index.html`, and `/about` should redirect to `/about/`). Do you know how to fix that?

Author: I updated the `.htaccess`, now both `/about.html` and `/about/` should redirect to `/about`.

About the `trailingSlash`:
I added the folder http://svelte-on-apache.great-site.net/slash (I just created that manually).
It's redirecting to http://svelte-on-apache.great-site.net/slash/.

Reviewer: Thank you — almost there! Just another couple of things I noticed:

* http://svelte-on-apache.great-site.net/slash/index.html redirects to http://svelte-on-apache.great-site.net/slash/index instead of http://svelte-on-apache.great-site.net/slash/ or 404ing
* http://svelte-on-apache.great-site.net/index.html redirects to http://svelte-on-apache.great-site.net/index instead of http://svelte-on-apache.great-site.net/ or 404ing
* http://svelte-on-apache.great-site.net/about/index.html redirects to http://svelte-on-apache.great-site.net/200 for some reason

Any idea if these are fixable?

Author: Hi, I updated the `.htaccess`. Here's the checklist, including recent fixes.

- [x] [/index.html or /index](http://svelte-on-apache.great-site.net/index.html) to /
- [x] [/about.html](http://svelte-on-apache.great-site.net/about.html) to /about
- [x] [/slash/index.html or /slash/index](http://svelte-on-apache.great-site.net/slash/index.html) to /slash/
- [x] [/slash/other.html](http://svelte-on-apache.great-site.net/slash/other.html) to /slash/other
- [x] [/slash/slash2/other.html](http://svelte-on-apache.great-site.net/slash/slash2/other.html) to /slash/slash2/other
- [x] `404`: [/slash/doesntexist.html](http://svelte-on-apache.great-site.net/slash/dontexist.html)
- [x] `404`: [/about/index.html](http://svelte-on-apache.great-site.net/about/index.html)
- [x] `404`: [/folder/doesnt/exist](http://svelte-on-apache.great-site.net/folder/doesnt/exist)
- [x] `404`: [/thispagedoesntexist](http://svelte-on-apache.great-site.net/thispagedoesntexist)

</IfModule>
```

> You can also create an `.htaccess` file in the static folder to automatically include the `.htaccess` on every build.
> 

Reviewer: ```suggestion
```
